internal/server/data/data.go

```diff
@@ -271,53 +310,44 @@ func Count[T models.Modelable](db *gorm.DB, selectors ...SelectorFunc) (int64, e
 	return count, nil
 }
 
-var infraProviderCache *models.Provider
-
 // InfraProvider is a lazy-loaded cached reference to the infra provider. The
 // cache lasts for the entire lifetime of the process, so any test or test
 // helper that calls InfraProvider must call InvalidateCache to clean up.
 func InfraProvider(db *gorm.DB) *models.Provider {
-	if infraProviderCache == nil {
-		infra, err := get[models.Provider](db, ByProviderKind(models.ProviderKindInfra))
-		if err != nil {
-			if errors.Is(err, internal.ErrNotFound) {
-				p := &models.Provider{Name: models.InternalInfraProviderName, Kind: models.ProviderKindInfra}
-				if err := add(db, p); err != nil {
-					logging.L.Panic().Err(err).Msg("failed to create infra provider")
-				}
-				return p
+	org := OrgFromContext(db.Statement.Context)
+	infra, err := get[models.Provider](db, ByProviderKind(models.ProviderKindInfra), ByOrgID(org.ID))
+	if err != nil {
+		if errors.Is(err, internal.ErrNotFound) {
+			p := &models.Provider{
```

Comment:
nit: does not have to be in this PR

Should we rely on the create function for this Infra provider creation?

Reply:
:100: https://github.com/infrahq/infra/pull/2902/commits/b1921154af3bd65039363123fafe72357ab31dfb (I see you found the TODO).